packages/@react-aria/toast/src/useToastRegion.ts

```diff
@@ -0,0 +1,82 @@
+import {AriaLabelingProps, DOMAttributes} from '@react-types/shared';
+import {focusWithoutScrolling} from '@react-aria/utils';
+import {getInteractionModality, useFocusWithin, useHover} from '@react-aria/interactions';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {RefObject, useEffect, useRef} from 'react';
+import {ToastState} from '@react-stately/toast';
+import {useLandmark} from '@react-aria/landmark';
+import {useLocalizedStringFormatter} from '@react-aria/i18n';
+
+export interface AriaToastRegionProps extends AriaLabelingProps {
+  /**
+   * An accessibility label for the toast region.
+   * @default "Notifications"
+   */
+  'aria-label'?: string
+}
+
+export interface ToastRegionAria {
+  /** Props for the landmark region element. */
+  regionProps: DOMAttributes
+}
+
+/**
+ * Provides the behavior and accessibility implementation for a toast region containing one or more toasts.
+ * Toasts are transient notifications of actions, errors, or other events in an application.
+ */
+export function useToastRegion<T>(props: AriaToastRegionProps, state: ToastState<T>, ref: RefObject<HTMLElement>): ToastRegionAria {
+  let stringFormatter = useLocalizedStringFormatter(intlMessages);
+  let {landmarkProps} = useLandmark({
+    role: 'region',
+    'aria-label': props['aria-label'] || stringFormatter.format('notifications')
+  }, ref);
+
+  let {hoverProps} = useHover({
+    onHoverStart: state.pauseAll,
+    onHoverEnd: state.resumeAll
+  });
+
+  let lastFocused = useRef(null);
+  let {focusWithinProps} = useFocusWithin({
+    onFocusWithin: (e) => {
+      state.pauseAll();
+      lastFocused.current = e.relatedTarget;
+    },
+    onBlurWithin: () => {
+      state.resumeAll();
+      lastFocused.current = null;
+    }
+  });
+
+  // When the region unmounts, restore focus to the last element that had focus
+  // before the user moved focus into the region.
+  // TODO: handle when the element has unmounted like FocusScope does?
+  // eslint-disable-next-line arrow-body-style
+  useEffect(() => {
```

Comment:
any way to let focus scope handle this?
I guess then the implementor would need to make a focus scope with `restore` and not contain

Or should the landmarks handle this?

Reply:
Well the problem is that we want it to restore to the last element that had focus before the user moved to the toast, not the element that had focus when the toast first mounted. So would probably require some changes to FocusScope - not sure if it's worth it unless we have another use case?

Reply:
Makes sense, thanks for explaining